swarm/pss/notify/notify.go

```diff
@@ -0,0 +1,276 @@
+package notify
+
+import (
+	"bytes"
+	"crypto/ecdsa"
+	"fmt"
+	"sync"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/p2p"
+	"github.com/ethereum/go-ethereum/rlp"
+	"github.com/ethereum/go-ethereum/swarm/pss"
+)
+
+const (
+	// sent from requester to updater to request start of notifications
+	MsgCodeStart = iota
+
+	// sent from updater to requester, contains a notification plus a new symkey to replace the old
+	MsgCodeNotifyWithKey
```

Comment:
how about we simplify this and only send the key here. 
I suggest getting rid of contentFunc callback in favour of a channel based auto-notify pattern.

Reply:
The idea was that we get notified of the current status immediately. Otherwise we will have to:

1) get ack from client when symkey is processed so we can send first update, or
2) wait for a "little while" then send the first update

I find:

1) unnecessary extra work and I don't really see the payoff.
2) not very nice

Reply:
but that situation is the same after you send the first update and about to call Notify again.
How do we know they received the first (with the key)? 
how about we send an asymmetrically encrypted start message from the subscriber containing a seed symmetric key which is then used in counter mode block cypher to send the updates?
lets have @nagydani input on this please

Reply:
Sure the client could generate it. Good idea.

Why do we need the extra complexity of counter mode?

Reply:
is "|" or or concat here?

Reply:
so now you would have to have transmission ordering then?

Reply:
So @nagydani do you have some input on this, please?